llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -11561,9 +11561,25 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
       unsigned OpIdx = isa<UnaryOperator>(VL0) ? 0 : 1;
       TTI::OperandValueInfo Op1Info = getOperandInfo(E->getOperand(0));
       TTI::OperandValueInfo Op2Info = getOperandInfo(E->getOperand(OpIdx));
-      return TTI->getArithmeticInstrCost(ShuffleOrOp, VecTy, CostKind, Op1Info,
-                                         Op2Info, {}, nullptr, TLI) +
-             CommonCost;
+      SmallVector<Value *, 16> Operands;
+      auto IsAllowedScalarTy = [](const Value *I) {
+        // Just to avoid regression with RISCV unittest.
+        return I->getType()->getPrimitiveSizeInBits() >= 32;
+      };
```

Comment:
Hacks are the bad way.

Reply:
These Args are being used by the  RISCV target for calculating the cast cost. To avoid miscalculation with the patch, had to use this hack.
If we dont want this, we need to introduce other parameter. There's no way out here.

Reply:
Or adjust RISCVTargetTransformInfo.cpp to handle it correctly, if there is something wrong, or update RISCV tests, if everything is fine

Reply:
adjusting RISCVTargetTransformInfo.cpp or updating the test & justifying the results is beyond my scope . w/o patch, casting is done only for scalar. w/ patch, its being done for all operands of the all the scalars.

Reply:
> adjusting RISCVTargetTransformInfo.cpp or updating the test & justifying the results is beyond my scope . w/o patch, casting is done only for scalar. w/ patch, its being done for all operands of the all the scalars.

In upstream you should care about all targets, you cannot introduce hacks just for your target. You can do this in your downstream version.

Reply:
Thats the reason I wanted to introduce one more API parameter

Reply:
I have tried to debug the RISCV backend and looking at the comment
`// f16 with zvfhmin and bf16 will be promoted to f32.`, calculating the cast cost for the args seems a valid choice which isnt happening currently.

But if this is incorrect, could you please help me to avoid this regression?